xarray/core/variable.py

```diff
@@ -933,9 +937,7 @@ def copy(self, deep=True, data=None):
                 data = indexing.MemoryCachedArray(data.array)
 
             if deep:
-                if hasattr(data, "__array_function__") or isinstance(
-                    data, dask_array_type
-                ):
+                if is_duck_array(data):
```

Comment:
Since dask arrays are duck arrays I don't see a need to explicitly check for dask here, unless I'm overlooking some details. Additionally, I believe this was discussed earlier and you suggested

> yes i think we want the proposed `is_duck_array`

So it should be okay with `is_duck_array`, right?

Reply:
I think the old check was from before `dask` had `__array_function__` (or because we had be backwards compatible with `dask` versions that didn't define `__array_function__`), so this should be fine.

Reply:
>  I believe this was discussed earlier and you suggested

haha sorry! thanks for clarifying